heartbeat/dovecot

```diff
@@ -0,0 +1,333 @@
+#!/bin/sh
+#
+# Resource script for Dovecot
+#
+# Description:  Manages Dovecot as an OCF resource in
+#               an high-availability setup.
+#
+# Author:       Raoul Bhatia <r.bhatia@ipax.at> : Original Author
+# License:      GNU General Public License (GPL)
+#
+#
+#       usage: $0 {start|stop|reload|monitor|validate-all|meta-data}
+#
+#       The "start" arg starts a Dovecot instance
+#
+#       The "stop" arg stops it.
+#
+# OCF parameters:
+#  OCF_RESKEY_binary
+#  OCF_RESKEY_config_file
+#  OCF_RESKEY_parameters
+#
+##########################################################################
+
+# Initialization:
+
+: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
+. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
+
+: ${OCF_RESKEY_binary="/usr/sbin/dovecot"}
+: ${OCF_RESKEY_config_file=""}
+: ${OCF_RESKEY_pid_file=""}
+: ${OCF_RESKEY_parameters=""}
+USAGE="Usage: $0 {start|stop|reload|monitor|validate-all|meta-data}";
+
+##########################################################################
+
+usage() {
+    echo $USAGE >&2
+}
+
+meta_data() {
+        cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="dovecot">
+<version>0.1</version>
+<longdesc lang="en">
+This script manages Dovecot as an OCF resource in a high-availability setup.
+</longdesc>
+<shortdesc lang="en">Manages a highly available Dovecot IMAP/POP3 server instance</shortdesc>
+
+<parameters>
+
+<parameter name="binary" unique="0" required="0">
+<longdesc lang="en">
+Full path to the Dovecot binary.
+For example, "/usr/sbin/dovecot".
+</longdesc>
+<shortdesc lang="en">Full path to Dovecot binary</shortdesc>
+<content type="string" default="/usr/sbin/dovecot" />
```

Comment:
You should use `_default` parameters in initialization, and reuse them here to avoid having to update it several places in the future.

Example:
https://github.com/ClusterLabs/resource-agents/blob/master/heartbeat/VirtualDomain#L20

Reply:
Like the other issues, it came from the postfix RA which doesn't use `_default` variables either.

Reply:
It's still a good idea to implement it with `_default` variables, and not add bad code practice in new agents.